### PR TITLE
Core/Misc: utf8 handle and utilities improvments and other misc stuff

### PR DESCRIPTION
**Pull request by Viste**
##### CHANGES PROPOSED:
-  replace AccountMgr::normalizeString with Utf8ToUpperOnlyLatin to prevent use non latin where must be only latin
-  add some helper functions for future and etc
-  use vector instead of list to contain Spell and Aura scripts and hooks ( need for #2254 )
-  added additional argument to Tokenizer class to make it behave like strtok - not returning empty tokens in case of multiple consecutive separators in input string
-  some renames from trinity to acore
-  Moved rng functions to separate header and added utility functions to select a random element from a container where each element can have different chance of being selected
- Update SFMT to latest
##### HOW TO TEST THE CHANGES:
build run be sure all is ok
##### Target branch(es):
- [x] Master

### CODE REVIEW

**Comment by Viste** [comment]
WIP still

File: src/common/Define.h [hunk 1]
```diff
@@ -1,11 +1,11 @@
 /*
- * Copyright (C) 2016+     AzerothCore <www.azerothcore.org>, released under GNU GPL v2 license: https://github.com/azerothcore/azerothcore-wotlk/blob/master/LICENSE-GPL2
+ * Copyright (C) 2016+     AzerothCore <www.azerothcore.org>, released under GNU AGPL3 v3 license: https://github.com/azerothcore/azerothcore-wotlk/blob/master/LICENSE-AGPL3
```
Comment (Winfidonarleyan) [inline src/common/Define.h]: old files at `GPL v2`

**Comment by Viste** [comment]
i think ready

**Comment by Viste** [comment]
Codacy is stupid asshole and lie about  \0-terminated look at code all is handled>_<

**Comment by Viste** [comment]
to many things what can fuckup many things open several small PRs

**Comment by FrancescoBorzi** [comment]
@Viste yes, smaller PRs are always more manageable :P

**Pull request closed without merging**